src/server/field.rs

```diff
@@ -714,4 +677,56 @@ impl<M: ReadEntry, Entry> ReadEntryResult<M, Entry> {
             io::Error::new(io::ErrorKind::InvalidData, msg),
         )
     }
-}
\ No newline at end of file
+}
+
+#[derive(Debug)]
+pub enum ParseHeaderError {
+    /// The header was not found
+    Missing,
```

Comment:
I'd like this variant to say which header was missing, though only `Content-Disposition` is required so maybe that can be hardcoded.

Also, this enum doesn't necessarily have to be public if it's converted to `io::Error` first, which ends up happening anyway.